### ISSUE TITLE
request: window local event subscription

### ISSUE THREAD

**Issue by aeosynth** [martanne/vis#439]
i want to listen to an event only for certain filetypes

**Comment by martanne** [comment]
For now something like the following should work?
```Lua
if vis.win.syntax ~= "your lexer name" then
    return
end
-- your event handler code here
```
Whether we should provide a custom API and how it would look like needs yet to be decided.

**Comment by aeosynth** [comment]
yes that would work, just asking for a nicer interface, ideally as simple as `win.events.subscribe` :thinking:
